### PR TITLE
Add 4.14 upgrade rollback job

### PR DESCRIPTION
**Pull request by shellyyang1989**
As 4.14.0-rc.0 is available, add 4.14 upgrade rollback job.

### CODE REVIEW

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 1]
```diff
@@ -58400,6 +58400,82 @@ periodics:
...
+  interval: 48h
+  labels:
+    ci-operator.openshift.io/cloud: aws
+    ci-operator.openshift.io/cloud-cluster-profile: aws-2
+    ci-operator.openshift.io/variant: nightly-4.14
+    ci.openshift.io/generator: prowgen
+    ci.openshift.io/no-builds: "true"
+    job-release: "4.14"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-upgrade-rollback-oldest-supported
```
Comment (wking) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:58421]: You'll want the usual `reporter_config` in here, right?  Compare #41525

Comment (wking) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:58421]: And also a 4.14 informer entry, along the lines of #39488?

Comment (shellyyang1989) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:58421]: Thank you Trevor! Added. PTAL

File: core-services/release-controller/_releases/release-ocp-4.14.json [hunk 2]
```diff
@@ -166,6 +166,11 @@
         "name": "periodic-ci-openshift-release-master-ci-4.14-e2e-aws-sdn-techpreview-serial"
       }
     },
+    "aws-ovn-upgrade-rollback": {
+      "prowJob": {
```
Comment (wking) [inline core-services/release-controller/_releases/release-ocp-4.14.json:170]: I think you want an `"optional": true,` in here to catch up with #43370.

Comment (shellyyang1989) [reply core-services/release-controller/_releases/release-ocp-4.14.json:171]: Good catch! Thank you!

**Comment by shellyyang1989** [comment]
cc @jianlinliu @jiajliu for review. Thank you

**Review by wking — APPROVED** [review]
/lgtm

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
The rehearsal passed. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/43401/rehearse-43401-periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-upgrade-rollback-oldest-supported/1706869502735552512

**Review by bradmwilliams — APPROVED** [review]

**Pull request merged**